server/src/main/java/io/druid/server/ClientInfoResource.java

```diff
@@ -300,6 +303,24 @@ public int compare(Interval o1, Interval o2)
     return metrics;
   }
 
+  @GET
+  @Path("/{dataSourceName}/candidates/intervals/{intervals}/numCandidates/{numCandidates}")
```

Comment:
Adding `@ResourceFilters(DatasourceResourceFilter.class)` to the endpoint will add the auth logic. So the endpoint will look like this -

```
  @GET
  @Path("/{dataSourceName}/candidates/intervals/{intervals}/numCandidates/{numCandidates}")
  @Produces(MediaType.APPLICATION_JSON)
  @ResourceFilters(DatasourceResourceFilter.class)
  public Iterable<LocatedSegmentDescriptor> getQueryTargets(
      @PathParam("dataSourceName") String datasource,
      @PathParam("intervals") String intervals,
      @PathParam("numCandidates") String numCandidates,
      @Context final HttpServletRequest req
  ) throws IOException
```
If this can be added then it's great otherwise I can do a follow up PR to add it.